### DISCUSSION TITLE
Katalog interfejsow.

### DISCUSSION THREAD

**Discussion by RKostrzynski** [AzureCommunityPL/group#380] [Pomoc]
Witam, podejrzewam ze nie jestem pierwsza osoba ktora probowala stworzyc katalog interfejsow w firmie. Jesli ktos ma sprawdzony wzor tego dukumentu to bardzo prosze sie nim podzielic. Dzieki.

**Comment by mgrabarz** [comment]
Macie w firmie jakiś API gateway albo chociaż SwaggerHub, czy ten katalog to ma powstać w Word lub Excel? ;)

**Comment by RKostrzynski** [comment]
Krotka odpowiedz to Nie. A dluga odpowiedz to, nawet gdyby byl to nie wiem czy dalbym rade zapisac tam komunikacje typu SFTP (lub inne nie API-jowe) polaczenia.

**Reply by mgrabarz** [reply]
Zdecydowanie nie opisałbyś SFTP za pomocą wspomnianych narzędzi. ;)

Niestety w kontekście papierowego "katalogu interfejsów" nie jestem w stanie poratować. Inna rzecz to sensowność robienia takich rzeczy na papierze - przecież to sztuka dla sztuki, tego się nie da na dłuższą metę utrzymać! Jedyny pomysł, który jakoś może się obronić to coś w stylu Enterprise Architect zapięty do repo ze współdzielonymi interfejsami rozwiązań i mapowanie użycia pomiędzy nimi.

**Reply by RKostrzynski** [reply]
To prawda, nie będzie łatwo utrzymywać takiego katalogu, ale jedyna alternatywa jest przymrużenie oka i udawanie ze nie wie co mamy w naszym portfolio. Plan długoterminowy jest taki, żeby zmodernizować większość interfejsów używając nowszych protokołów, ale żeby to zrobić musimy wiedzieć co mamy w naszym „arsenale” na dzień dzisiejszy. Dzięki za odpowiedz i pozdrawiam.

**Reply by kaluzaaa** [reply]
@RKostrzynski podpowiedź - API wychodzi z katalogu aplikacji z podziałem na downstream/upstream danych.
